src/kldiv.jl

```diff
@@ -0,0 +1,31 @@
+function KL(p::AbstractMvNormal, q::AbstractMvNormal)
+    # This is the generic implementation for AbstractMvNormal, you might need to specialize for your type
+    length(p) == length(q) || throw(
+        DimensionMismatch(
+            "Distributions p and q have different dimensions $(length(p)) and $(length(q))",
+        ),
+    )
+    Σp, Σq = cov.((p, q))
+    Δμ = mean(p) - mean(q)
+    return 0.5 * (tr(Σq \ Σp) + dot(Δμ, Σq \ Δμ) - length(p) + logdet(Σq) - logdet(Σp))
+end
+
+function KL(p::MvNormal, q::MvNormal)
+    # We use p.Σ and q.Σ to take the advantage that they are defined as PDMats objects
+    length(p) == length(q) || throw(
+        DimensionMismatch(
+            "Distributions p and q have different dimensions $(length(p)) and $(length(q))",
+        ),
+    )
+    return 0.5 * (
+        tr(q.Σ \ p.Σ) + invquad(q.Σ, mean(p) - mean(q)) - length(p) + logdet(q.Σ) -
+        logdet(p.Σ)
+    )
+end
+
+kldivergence(p, q) = KL(p, q)
```

Comment:
Shouldn't this be defined in Distributions? It seems a bit weird for the code to live in this package. Also it seems you shadow `StatsBase.kldivergence`?

(BTW I would also recommend to not use broadcasting - just write `cov(p)` and `cov(q)` explicitly - and to not use Float64 literals)

Reply:
Yes, it would be great if this was defined in Distributions. I've copied this from @theogf's KLDivergences package because it had some AD issues (and to reduce the number of dependencies a bit). Will be very happy to remove this again once Distributions supports it out of the box.